src/annotator/integrations/html.ts

```diff
@@ -22,6 +22,43 @@ import {
 // content line lengths and scale are too short to be readable.
 const MIN_HTML_WIDTH = 480;
 
+/**
+ * `auto`: The client will decide if side-by-side is enabled. If enabled, it
+ *         will apply some heuristic to determine how the content is affected.
+ *         This is default value.
+ * `manual`: The host app wants to manually take full control of side-by-side,
+ *           effectively disabling the logic in client.
+ */
+type SideBySideMode = 'auto' | 'manual';
+
+type SideBySideOptions = {
+  mode: SideBySideMode;
+};
+
+function initialSideBySideMode(
+  enabled: boolean,
+  defaultValue?: SideBySideMode
+): SideBySideMode {
+  if (!enabled) {
+    // Manual mode effectively disables side-by-side
+    return 'manual';
+  }
+
+  return defaultValue ?? 'auto';
+}
+
+function sideBySideModeOnFlagChange(
+  enabled: boolean,
+  initiallyProvidedMode?: SideBySideMode
+): SideBySideMode {
+  // If the mode was explicitly provided as `manual`, keep it that way
+  if (initiallyProvidedMode === 'manual') {
+    return 'manual';
+  }
+
+  return enabled ? 'auto' : 'manual';
+}
```

Comment:
I'm thinking the `SideBySideMode` type and these two helper functions might go into `html-side-by-side` module.